### PR TITLE
Remove tensor subclass detection logic from weights_only unpickler

### PR DESCRIPTION
**Pull request by mikaylagawarecki**
Remove logic to auto-detect and allow subclasses that did not override certain methods from the weights_only unpickler from https://github.com/pytorch/pytorch/pull/124331 for 2.4 release

Subclasses should be loadable using `torch.serialization.add_safe_globals`

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #127939
* __->__ #127808

cc @albanD

### CODE REVIEW

**Review by malfet — APPROVED** [review]
LGTM, but it would be nice to have at least one test that validate subclasses work via globals. Also, perhaps worth having a safe globals context manager

File: test/test_serialization.py [hunk 1]
```diff
@@ -4207,93 +4189,6 @@ def test_empty_class_serialization(self):
...
-                    else:
-                        # When the given methods are all back to default, weights_only load should finally work
-                        loaded = torch.load(f, weights_only=True)
-            finally:
-                for method, func in restore_methods.items():
-                    setattr(subclass, method, func)
-                a = subclass(torch.randn(2, 3))
-
     @skipIfTorchDynamo("name 'SYNTHETIC_LOCAL' is not defined")
     def test_safe_globals_for_weights_only(self):
```
Comment (mikaylagawarecki) [inline test/test_serialization.py:4193]: @malfet This tests that functionality with `TwoTensor`, is that what you were looking for?

**Pull request closed without merging**